### PR TITLE
VST: LockBit (Attack module)

### PR DESCRIPTION
**Pull request by gerbsec**

### CODE REVIEW

File: endpoint/attack/go.mod [hunk 1]
```diff
@@ -0,0 +1,3 @@
+module github.com/preludeorg/test/endpoint/attack
```
Comment (privateducky) [inline endpoint/attack/go.mod:1]: should we make this a module or package?

File: endpoint/attack/attack.go [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+	"darwin":  {"bash", "-c"},
+}
+
+func T1082() []string {
+	var cmd = map[string]string{
+		"windows": "systeminfo",
+		"linux":   "hostnamectl",
+		"darwin":  "sysctl -a -h",
+	}
+	command := append(supported[runtime.GOOS], cmd[runtime.GOOS])
```
Comment (privateducky) [inline endpoint/attack/attack.go:27]: in cases like these, just return the thing vs setting it as a variable and then returning

File: endpoint/attack/go.mod (hunk 1 above)
Comment (gerbsec) [reply endpoint/attack/go.mod:1]: package is fine, didn't mean to push that.

File: endpoint/attack/attack.go (hunk 2 above)
Comment (gerbsec) [reply endpoint/attack/attack.go:27]: im about to push something better i think

**Review by privateducky — COMMENTED** [review]
I'd like to see how the first 2 emerging threat (ET) tests go before committing to an attack.go direction (which I think is inevitable). 

reason: I'm not sure where we'll naturally fall with the mix of shell vs go vs relying on embed to drop/start tests in other languages. 

considering there are many ways to do T1082 (for example), if this implementation gets flagged it would kill all associated tests by default vs if each test implemented their own version. almost the opposite of what a SW eng thinks about!

**Comment by gerbsec** [comment]
> I'd like to see how the first 2 emerging threat (ET) tests go before committing to an attack.go direction (which I think is inevitable).
> 
> reason: I'm not sure where we'll naturally fall with the mix of shell vs go vs relying on embed to drop/start tests in other languages.
> 
> considering there are many ways to do T1082 (for example), if this implementation gets flagged it would kill all associated tests by default vs if each test implemented their own version. almost the opposite of what a SW eng thinks about!

check out the changes i just pushed up rn

**Comment by gerbsec** [comment]
> I'd like to see how the first 2 emerging threat (ET) tests go before committing to an attack.go direction (which I think is inevitable).
> 
> reason: I'm not sure where we'll naturally fall with the mix of shell vs go vs relying on embed to drop/start tests in other languages.
> 
> considering there are many ways to do T1082 (for example), if this implementation gets flagged it would kill all associated tests by default vs if each test implemented their own version. almost the opposite of what a SW eng thinks about!

I think in an attack.go module/package we can easily create variants of the same t code whereas if I was to create a singular VST ( see #120 ) then if that gets flagged, we'll need to go in and change every single VST and it's implementation. I gave an example with the _DES encryption, I alternated to a different encryption type to demonstrate how easy it is to add a variant.

**Comment by bfuzzy1** [comment]
My thoughts on this:

Having an Attack package where we primarily call Endpoint seems weighty to me. What I mean by that is each Tcode function should be generalized enough to where we could swap things out per VST without impacting the rest of the tests in any way. One approach which you have in this is to create different functions for each variation of a single Tcode, which is not a bad approach but we will quickly run into dozens of function variations and a massive attack package which we then need to maintain.

**Pull request closed without merging**